### PR TITLE
cmd: ExecInCoreSnap looks in "core" snap first, and only in "ubuntu-core" snap if rev>125.

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -34,7 +34,18 @@ import (
...
+// newCore is the place to look for the core snap; everything in this
+// location will be new enough to re-exec into.
+const newCore = "/snap/core/current"
+
+// oldCore is the previous location of the core snap. Only things
+// newer than minOldRevno will be ok to re-exec into.
+const oldCore = "/snap/ubuntu-core/current"
+
+// old ubuntu-core snaps older than this aren't suitable targets for re-execage
+const minOldRevno = 126
```
Comment (chipaca) [inline cmd/cmd.go]: `// chosen by fair dice roll.`

File: cmd/cmd.go [hunk 2]
```diff
@@ -34,8 +35,19 @@ import (
 // to be set to 1 (do re-exec); that is: set it to 0 to disable.
 const key = "SNAP_REEXEC"
 
-// SwitchToTheRealOne makes sure you're executing the "real"
-// binary. I.e. the one that ships in the ubuntu-core snap.
+// newCore is the place to look for the core snap; everything in this
+// location will be new enough to re-exec into.
+const newCore = "/snap/core/current"
```
Comment (zyga) [inline cmd/cmd.go]: Because otherwise we will still run snaps with the ubuntu-core bind mounts

**Comment by zyga** [comment]
+1

File: cmd/cmd.go (hunk 2 above)
Comment (chipaca) [inline cmd/cmd.go]: retest this please
ya ol' rustbucket

**Comment by chipaca** [comment]
retest this please
ya ol' rustbucket

**Pull request merged**